### ISSUE TITLE
Custom HLS stream not playing on Oculus Quest

### ISSUE THREAD

**Issue by ElHacker** [RenderHeads/UnityPlugin-AVProVideo#472]
**Describe the bug**
I've configured my Quest Android app to use Fast OES Path. I'm using AvProVideo/Unlit/Opaque (texture+color support) - Android OES ONLY shader.

The app does play video when I consume this test link https://stream.mux.com/Wxle5yzErvilJ02C13zuv8OSeROvfwsjS.m3u8

But when I use my custom HLS service I set up myself to do live streaming the video doesn't show in the Android build, I only get a black screen. Although It works fine in the Unity Editor Windows 10, I can also play my stream without problem on VLC player.

I'm starting to wonder whether this has to do with the way I'm encoding the stream with ffmpeg on the streaming service, but I don't know for sure. 

This is what I'm currently doing:
`ffmpeg -i pipe:0 -preset ultrafast -tune zerolatency -fflags nobuffer -vcodec libx264 -c:a aac -b:a 128k -r 30 -g 30 -b:v 10M -probesize 32 -f flv rtmp://myserver`

Other notes:
My HLS service is served via HTTPS, so I don't have the cleartext issue reported by other people here.
I also verified I have the Internet Required permission for my build.
Video API: Exo Player

1. Do you have a set of recommended practices to do encoding?
2. Anywhere else I should be looking in my AV Pro config for Android?

**Your Setup (please complete the following information):**
 - Unity version: 2020.1.0f1
 - AVPro Video version: 1.11.4 / I've paid for the Android license.
 - Operating system version: Android for Oculus Quest (not sure about the android version)
 - Device model: Oculus Quest
 - Video specs (resolution, frame-rate, codec, file size):
1280x720
30 fps
libx264
live streaming (assume infinite file size)

**Comment by AndrewRH** [comment]
Hi,

It's tricky to work out what the issue is, but I would recommend starting with the Android log file.  You can view this through adb logcat or Android Studio or another tool.  There will be a lot of information in the log, but somewhere in there it would have an error message explaining why the connection couldn't be made.

Another thing to try is adding "-pix_fmt yuv420p" to your ffmpeg command as I think this may be needed for be compatibility.

Thanks,

**Comment by ElHacker** [comment]
Hey Andrew thanks for the help in encoding I'll take a look at it and see if it improves things on my custom stream.

In case anyone else is facing the same problem:
The way I worked around the issue was by retrying to play to my HLS stream until it was successful.
I made use of the `mediaPlayer.Events.AddListener(OnVideoEvent)`

Then in `OnVideoEvent `method I checked for the `EventType.Error` (it usually triggers on stream loading errors) and just retry to load the video with `mediaPlayer.OpenVideoFromFile` method.

I know this is not ideal, but at least it works.
